modelon/impact/client/entities/experiment.py

```diff
@@ -705,7 +715,9 @@ def _get_custom_function(self, analysis: Dict[str, Any]) -> CustomFunction:
         custom_function_meta = self._sal.custom_function.custom_function_get(
             self._workspace_id, self.custom_function
         )
-        custom_function_params = analysis["parameters"]
+        custom_function_params = {
+            param["name"]: param["value"] for param in analysis["parameters"]
```

Comment:
analysis["parameters"] -> analysis.get("parameters", []) as above? Also, consider breaking this parsing out?